File: frontend/types/support.ts
export interface SupportCreate {
  person_name: string;
  subject: string;
  message: string;
}

export interface Support {
  id: number;
  clinic_id: number;
  clinic_name?: string;
  person_name: string;
  email?: string;
  phone?: string;
  subject: string;
  message: string;
  status: number;
  resolved_at?: string;
  created_at: string;
}

File: backend/app/schemas/support.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SupportCreate(BaseModel):
    person_name: str
    subject: str
    message: str


class SupportResponse(BaseModel):
    id: int
    clinic_id: int
    clinic_name: Optional[str] = None
    person_name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    subject: str
    message: str
    status: int = 0
    resolved_at: Optional[datetime] = None
    created_at: datetime

    class Config:
        from_attributes = True
